sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3227,6 +3227,33 @@ object GpuOverrides extends Logging {
           ParamCheck("regexp", TypeSig.lit(TypeEnum.STRING), TypeSig.STRING),
           ParamCheck("idx", TypeSig.lit(TypeEnum.INT), TypeSig.INT))),
       (a, conf, p, r) => new GpuRegExpExtractAllMeta(a, conf, p, r)),
+    expr[ParseUrl](
+      "Extracts a part from a URL",
+      ExprChecks.projectOnly(TypeSig.STRING, TypeSig.STRING,
+        Seq(ParamCheck("url", TypeSig.STRING, TypeSig.STRING),
+          ParamCheck("partToExtract", TypeSig.lit(TypeEnum.STRING), TypeSig.STRING)),
+          // Should really be an OptionalParam
+          Some(RepeatingParamCheck("key", TypeSig.lit(TypeEnum.STRING), TypeSig.STRING))),
+      (a, conf, p, r) => new ExprMeta[ParseUrl](a, conf, p, r) {
+        val failOnError = a.failOnError
+
+        override def tagExprForGpu(): Unit = {
+          if (failOnError) {
+            willNotWorkOnGpu("Fail on error is not supported on GPU when parsing urls.")
+          }
+          val partToExtract = childExprs(1).convertToGpu()
```

Comment:
Thais is not guaranteed to work. You said it was a `lit`, but `lit`is more lenient that just a Literal. please use `extractStringLit` instead. 

https://github.com/NVIDIA/spark-rapids/blob/ed3a1f3e879abd7b0574cbeb990eaa8f421ea47d/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala#L591

i.e.

```
extractStringLit(childExprs(1)).map(_.toUpperCase) match {
  case Some("PROTOCOL") =>
  case Some(other) =>
    willNotWorkOnGpu(s"Part to extract $other is not supported on GPU")
  case None =>
    // Should never get here, but just in case
    willNotWorkOnGpu("GPU only supports a literal for the part to extract")
}
```